### PR TITLE
Adding a new API Proposal Issue  Template

### PR DESCRIPTION
**Pull request by mkArtakMSFT**
I'm copying the existing api-proposal template from the dotnet/runtime repo with very light modifications to align with aspnetcore repo standards.

This was asked by @davidfowl who is member of the @dotnet/aspnet-api-review  team.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/25_api_proposal.md
Comment (pranavkm) [inline .github/ISSUE_TEMPLATE/25_api_proposal.md:24]: Should we use ASP.NET Core specific examples? Also what's the difference between this template and a design proposal? How is a contributor supposed to figure out what works best (because an API usually doesn't live in isolation without a design that goes with it)?

Comment (davidfowl) [reply .github/ISSUE_TEMPLATE/25_api_proposal.md:24]: > Should we use ASP.NET Core specific examples?
Yes

> How is a contributor supposed to figure out what works best (because an API usually doesn't live in isolation without a design that goes with it)?

This is a good point but the design proposal is heavy for adding scenario based APIs. If you can't describe it in a single issue like this then it's a good candidate for a design proposal. This is for adding APIs like:

- https://github.com/dotnet/aspnetcore/issues/5877
- https://github.com/dotnet/aspnetcore/issues/12978
- https://github.com/dotnet/aspnetcore/issues/13270
- https://github.com/dotnet/aspnetcore/issues/17972

Etc

Comment (mkArtakMSFT) [reply .github/ISSUE_TEMPLATE/25_api_proposal.md:24]: Can you guys please provide good examples to include then?

File: .github/ISSUE_TEMPLATE/25_api_proposal.md
Comment (ryan-dayrit) [inline .github/ISSUE_TEMPLATE/25_api_proposal.md:17]: Since you used `Alternative Designs` verbiage, should this be `Proposed Design`?

Comment (mkArtakMSFT) [reply .github/ISSUE_TEMPLATE/25_api_proposal.md:17]: Thanks for your comment, @ryan-dayrit.
This is actually about the other designs the owner of the proposal may have considered before proposing this design. Hence, this phrase seems valid in this context.

File: .github/ISSUE_TEMPLATE/25_api_proposal.md
Comment (mkArtakMSFT) [reply .github/ISSUE_TEMPLATE/25_api_proposal.md:24]: @davidfowl, @pranavkm I'm waiting for relevant examples here before I merge.

File: .github/ISSUE_TEMPLATE/25_api_proposal.md
Comment (davidfowl) [inline .github/ISSUE_TEMPLATE/25_api_proposal.md:26]:
```suggestion
namespace Microsoft.AspNetCore.Http
{
    public static class HttpResponseWritingExtensions
    {
+       public Task WriteAsync(this HttpResponse response, StringBuilder builder);
    }
```

File: .github/ISSUE_TEMPLATE/25_api_proposal.md
Comment (davidfowl) [reply .github/ISSUE_TEMPLATE/25_api_proposal.md:24]: Added a suggestion

**Review by pranavkm — APPROVED** [review]

**Review by davidfowl — APPROVED** [review]

**Pull request merged**
